### PR TITLE
bug: fix handle leaks in Windows process functions

### PR DESCRIPTION
**Pull request by muffins**
This addresses #4322 and #4317, as well as adds a getter for the current process pid.

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit a1ace6ffe5005ba4dc4d5756678f728d9e3b0380 (Job results: [2939](https://jenkins.osquery.io/job/osqueryPullRequestOSX/2939/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit a1ace6ffe5005ba4dc4d5756678f728d9e3b0380 (Job results: [2940](https://jenkins.osquery.io/job/osqueryPullRequestOSX/2940/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit a1ace6ffe5005ba4dc4d5756678f728d9e3b0380 (Job results: [3814](https://jenkins.osquery.io/job/osqueryPullRequestWindows/3814/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit a1ace6ffe5005ba4dc4d5756678f728d9e3b0380 (Job results: [6927](https://jenkins.osquery.io/job/osqueryPullRequestLinux/6927/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit a1ace6ffe5005ba4dc4d5756678f728d9e3b0380 (Job results: [3815](https://jenkins.osquery.io/job/osqueryPullRequestWindows/3815/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit a1ace6ffe5005ba4dc4d5756678f728d9e3b0380 (Job results: [1366](https://jenkins.osquery.io/job/osqueryPullRequestFreeBSD/1366/)) failed one or more tests (**FreeBSD**).

**Comment by osqueryer** [comment]
:-1: The commit a1ace6ffe5005ba4dc4d5756678f728d9e3b0380 (Job results: [6928](https://jenkins.osquery.io/job/osqueryPullRequestLinux/6928/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit a1ace6ffe5005ba4dc4d5756678f728d9e3b0380 (Job results: [1367](https://jenkins.osquery.io/job/osqueryPullRequestFreeBSD/1367/)) failed one or more tests (**FreeBSD**).

File: osquery/core/system.cpp [hunk 1]
```diff
@@ -372,8 +372,7 @@ Status createPidFile() {
   }
 
   // If no pidfile exists or the existing pid was stale, write, log, and run.
-  auto pid = boost::lexical_cast<std::string>(
-      PlatformProcess::getCurrentProcess()->pid());
+  auto pid = boost::lexical_cast<std::string>(PlatformProcess::getCurrentPid());
```
Comment (theopolis) [inline osquery/core/system.cpp]: Nit: We found that std::to_string is faster and easier than a boost lexical cast.

**Pull request merged**